src/components/IdentityVerificationProcess.tsx

```diff
@@ -39,54 +64,75 @@ const IdentityVerificationProcess = () => {
     });
   };
 
-  const handleProceed = () => {
-    setStage(2);
-  };
 
-  const handleSelectAccount = (account) => {
-    console.log(`Selected account: ${account}`);
-    // Implement account selection logic here
-  };
+  const handleBack = () => { setStage(n => n > 0 ? n - 1 : n); };
+  const handleProceed = () => { setStage(n => n < 2 ? n + 1 : n); };
 
-  const handleRemoveIdentity = () => {
-    console.log('Removing identity');
-    // Implement identity removal logic here
-  };
-
-  const handleLogout = () => {
-    console.log('Logging out');
-    // Implement logout logic here
-  };
-
-  const renderStage = () => {
+  const StageContent = () => {
     switch(stage) {
-      case 0:
-        return <IdentityForm />;
-      case 1:
+      case Stages.SetIdentityForm:
+        return <IdentityForm handleProceed={handleProceed} />;
+      case Stages.Challenges:
         return <ChallengeVerification
           onVerify={handleVerifyChallenge}
           onCancel={handleCancel}
           onProceed={handleProceed}
         />;
-      case 2:
+      case Stages.Congrats:
         return <CompletionPage />;
       default:
         return null;
     }
   };
+  
+  useEffect(() => {
+    if (appStateSnap.account && import.meta.env.DEV) {
+      const _ss58Info = getSs58AddressInfo(appStateSnap.account.address);
+      console.log({ ss58Info: _ss58Info, })
+    }
+  }, [appStateSnap.account]) 
 
-  const appStateSnap = useSnapshot(appState)
-  if (appStateSnap.account) {
-    return (
-      <SignerProvider signer={appStateSnap.account?.polkadotSigner}>
-        <div className="w-full max-w-3xl mx-auto p-6 bg-white border border-stone-300">
-          <Header />
-          <ProgressBar progress={stage === 0 ? 0 : stage === 1 ? 50 : 100} />
-          {renderStage()}
+  const percentage = appStateSnap.verificationProgress / (Object.keys(IdentityVerificationStates).length / 2 -2) * 100
+  useEffect(() => console.log({ percentage, 
+    value: appStateSnap.verificationProgress,
+    key: IdentityVerificationStates[appStateSnap.verificationProgress],
+  }), [percentage])
+
+  const canProceed = useMemo((() => 
+    (stage <= Stages.SetIdentityForm 
+      && appStateSnap.verificationProgress >= IdentityVerificationStates.JudgementRequested
+    )
+    || (stage <= Stages.Challenges 
+      && appStateSnap.verificationProgress >= IdentityVerificationStates.IdentityVerifid  
```

Comment:
IdentityVerifid 

typo

Reply:
Corrected in commit 111ecd16674c163a521f73d7c0af0f6450a449d7